### PR TITLE
Use local dict instead of modifying locals()

### LINKED ISSUE #246: py3.13 - `fmf ls --condition 'float(release) >= 7'` broken
Used in `tests/unit/test_cli.py` - `test_conditions`. Can be reproduced easily:
```
$ cd examples/conditions
$ fmf ls --condition 'flaot(release) >= 7'
```

On python3.12 and older it prints 3 tests. On python3.13 the output is empty.

Blocks fedora rawhide - https://bugzilla.redhat.com/show_bug.cgi?id=2279994

### PR DESCRIPTION
**Pull request by martinhoyer**
Fixes #246. At least it does for me locally.

### CODE REVIEW

**Comment by martinhoyer** [comment]
Ah, I've duplicated https://github.com/teemtee/fmf/pull/247, lol

**Comment by lukaszachy** [comment]
I wonder why my didn't compile on rawhide but yours did ;)

**Review by lukaszachy — APPROVED** [review]

**Comment by lukaszachy** [comment]
And rawhide wants to install tmt, which it can't.. I'd say lets ignore.

File: fmf/utils.py [hunk 1]
```diff
@@ -202,9 +202,10 @@ def evaluate(expression, data, _node=None):
     Expects data dictionary which will be used to populate local
     namespace. Used to provide flexible conditions for filtering.
     """
-    locals().update(data)
+    namespace = dict()
+    namespace.update(data)
```
[the remark this answers is not in the document]
Comment (lukaszachy) [reply fmf/utils.py:206]: Better to have a copy. "eval" can do anything to the original `data`.

At least for me this is more readable.

[the remark this answers is not in the document]
Comment (martinhoyer) [reply fmf/utils.py]: oh, just pushed https://github.com/teemtee/fmf/pull/248/commits/8be353678877405a5025907539053293eae1326a before reading Lukáš's comment. So we go with copy?

[the remark this answers is not in the document]
Comment (happz) [reply fmf/utils.py]: `copy` or `dict(data)`, both should produce new dictionary instance.

[the remark this answers is not in the document]
Comment (martinhoyer) [reply fmf/utils.py]: `dict(data)` is what is there now. @lukaszachy ok with you?

**Review by happz — APPROVED** [review]

**Pull request merged**